k8s/cmd/commands/sidecar/inject_test.go

```diff
@@ -0,0 +1,1057 @@
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := GetWorkloadObj(tt.args.workload)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GetWorkloadObj() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				//fmt.Println(got)
+				//fmt.Println(tt.want)
+				t.Errorf("GetWorkloadObj() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+/*func TestGetManifestFromTemplate(t *testing.T) {
+	flags.WorkloadYaml = "workload.yaml"
+	resource, err := getWorkloadResource()
+	if err != nil {
+		log.Fatal(err, "failed to get the workload resource")
+	}
+	//sidecar, err := buildSidecar("vineyard-system")
+	//fmt.Println(sidecar)
+	//workloadObj, err := GetWorkloadObj(resource)
+	//fmt.Println(workloadObj)
+	_, err = GetManifestFromTemplate(resource)
+
+	require.NoError(t, err)
+
+}*/
+
+func TestGetManifestFromTemplate(t *testing.T) {
+	resource := `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  namespace: vineyard-system
+spec:
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        ports:
+        - containerPort: 80
+`
+	type args struct {
+		workload string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    OutputManifests
+		wantErr bool
+	}{
+		// TODO: Add test cases.
+		{
+			name: "Test case",
+			args: args{workload: resource},
+			want: OutputManifests{
+				Workload:    `{"apiVersion":"apps/v1","kind":"Deployment","metadata":{"creationTimestamp":null,"name":"nginx-deployment","namespace":"vineyard-system","ownerReferences":[]},"spec":{"selector":{"matchLabels":{"app":"nginx"}},"strategy":{},"template":{"metadata":{"creationTimestamp":null,"labels":{"app":"nginx","app.vineyard.io/name":"vineyard-sidecar","app.vineyard.io/role":"vineyardd"}},"spec":{"containers":[{"command":null,"image":"nginx:1.14.2","name":"nginx","ports":[{"containerPort":80}],"resources":{},"volumeMounts":[{"mountPath":"/var/run","name":"vineyard-socket"}]},{"command":["/bin/bash","-c","/usr/bin/wait-for-it.sh -t 60 vineyard-sidecar-etcd-service.vineyard-system.svc.cluster.local:2379; sleep 1; /usr/local/bin/vineyardd --sync_crds true --socket /var/run/vineyard.sock --size  --stream_threshold 80 --etcd_cmd etcd --etcd_prefix /vineyard --etcd_endpoint http://vineyard-sidecar-etcd-service:2379\n"],"env":[{"name":"VINEYARDD_UID","value":null},{"name":"VINEYARDD_NAME","value":"vineyard-sidecar"},{"name":"VINEYARDD_NAMESPACE","value":"vineyard-system"}],"image":"vineyardcloudnative/vineyardd:latest","imagePullPolicy":"IfNotPresent","name":"vineyard-sidecar","ports":[{"containerPort":9600,"name":"vineyard-rpc","protocol":"TCP"}],"resources":{"limits":null,"requests":null},"volumeMounts":[{"mountPath":"/var/run","name":"vineyard-socket"}]}],"volumes":[{"emptyDir":{},"name":"vineyard-socket"}]}}},"status":{}}` + "\n",
```

Comment:
It's hard to maintain the long string, a possible way is to output the string as YAML.